narayana/scripts/hudson/jenkins.sh

```diff
@@ -83,11 +83,9 @@ fi
 
 for i in $THREAD_COUNTS
 do
-  if [ -z "${JMHARGS}" ] ; then
-    JMHARGS="-t $i -r 25 -f 2 -wi 5 -i 5"
-  else
-    JMHARGS="-t $i -r 20 -f 1 -wi 3 -i 5"
-  fi
+#  if [ -z "${JMHARGS}" ] ; then
+    JMHARGS="-t $i -r 30 -f 3 -wi 5 -i 5"
```

Comment:
We need to allow the caller to be able to set JMARGS so only set the variable if it is unset. So just remove the else part of the if statement:

```bash
  if [ -z "${JMHARGS}" ] ; then 
    JMHARGS="-t $1 -r 25 -f 2 -wi 5 -i 5"           
  fi
```

Reply:
here if we do what you mentioned, then the $1 i.e. the thread count will never change. It would be fixed.
Is here anything, that I am missing to understand?

Reply:
I am unsure why we allow the CI job to tune the JMH parameters but I think we should still allow it and just set the thread count:

```bash
for i in $THREAD_COUNTS
do
  if [ -z "${JMHARGS}" ] ; then
    JMHARGS="-r 25 -f 2 -wi 5 -i 5"
  fi
  JMHARGS="-t $i $JMHARGS" bm bm-output.txt $COMPARISON $COMPARISON_COUNT
done
```

Reply:
In fact, check for JMARGS outside of the loop:
```bash
if [ -z "${JMHARGS}" ] ; then
  JMHARGS="-r 25 -f 2 -wi 5 -i 5"
fi
for i in $THREAD_COUNTS
do
  JMHARGS="-t $i $JMHARGS" bm bm-output.txt $COMPARISON $COMPARISON_COUNT
done
```

Reply:
Ok, sure will do that

Reply:
But I think this solution has a small problem.
Suppose : THREAD_COUNTS="1 10 50 100 300 400"

now if we run the above code:
we get - JMHARGS = -t 400 -t 300 -t 100 -t 50 -t 10 -t 1 -r 25 -f 2 -wi 5 -i 5